packages/botfuel-dialog/src/brains/brain.js

```diff
@@ -14,6 +14,11 @@
  * limitations under the License.
  */
 
+// @flow
+
+import type { Config } from '../config';
+import type { DialogsData, UserData, ConversationData } from '../types';
```

Comment:
Do we really want to create all these extra types?

Reply:
I think that these types improve readability and they are of great use when we need to change things.

Moreover, they make more apparent situations where the data manipulations are unclear and error prone (so they suggest good places for refactorings).